src/webhooks/gocd/gocd.ts

```diff
@@ -1,12 +1,37 @@
-import { FastifyRequest } from 'fastify';
+import * as Sentry from '@sentry/node';
+import { FastifyReply, FastifyRequest } from 'fastify';
 
 import { GoCDResponse } from '@types';
 
+import { GOCD_WEBHOOK_SECRET } from '@/config';
+import { extractAndVerifySignature } from '@/utils/extractAndVerifySignature';
 import { gocdevents } from '@api/gocdevents';
 
-export async function handler(request: FastifyRequest<{ Body: GoCDResponse }>) {
-  return {};
-  const { body }: { body: GoCDResponse } = request;
-  gocdevents.emit(body.type, body);
-  return {};
+export async function handler(
+  request: FastifyRequest<{ Body: GoCDResponse }>,
+  reply: FastifyReply
+) {
+  // If the webhook secret is not defined, return a 500
+  if (GOCD_WEBHOOK_SECRET === undefined) {
+    return reply.code(500).send();
```

Comment:
should we send a Sentry error if this happens for visbility?

Reply:
Thinking it may be hard to track down if we just return a 500 http response

Reply:
Moved it into the try catch and threw an error that will be caught by sentry